subxt/src/client/lightclient/builder.rs

```diff
@@ -0,0 +1,197 @@
+    /// to fail, feel free to pass `u32::max_value()`.
+    ///
+    /// Default is 128.
+    pub fn max_pending_requests(mut self, max_pending_requests: NonZeroU32) -> Self {
+        self.max_pending_requests = max_pending_requests;
+        self
+    }
+
+    /// Maximum number of active subscriptions before new ones are automatically
+    /// rejected. Any JSON-RPC request that causes the server to generate notifications
+    /// counts as a subscription.
+    ///
+    /// Default is 1024.
+    pub fn max_subscriptions(mut self, max_subscriptions: u32) -> Self {
+        self.max_subscriptions = max_subscriptions;
+        self
+    }
+
+    /// If the chain spec defines a parachain, contains the list of relay chains to choose
+    /// from. Ignored if not a parachain.
+    ///
+    /// This field is necessary because multiple different chain can have the same identity.
+    ///
+    /// For example: if user A adds a chain named "Kusama", then user B adds a different chain
+    /// also named "Kusama", then user B adds a parachain whose relay chain is "Kusama", it would
+    /// be wrong to connect to the "Kusama" created by user A.
+    pub fn potential_relay_chains(
+        mut self,
+        potential_relay_chains: impl IntoIterator<Item = ChainId>,
+    ) -> Self {
+        self.potential_relay_chains = Some(potential_relay_chains.into_iter().collect());
+        self
+    }
+
+    /// Build the light client with specified URL to connect to.
+    /// You must provide the port number in the URL.
+    ///
+    /// ## Panics
+    ///
+    /// Panics if being called outside of `tokio` runtime context.
+    #[cfg(feature = "jsonrpsee")]
+    pub async fn build_from_url<T: Config, Url: AsRef<str>>(
+        self,
+        url: Url,
+    ) -> Result<LightClient<T>, Error> {
+        let chain_spec = fetch_url(url.as_ref()).await?;
+
+        self.build_client(chain_spec).await
+    }
+
+    /// Build the light client from chain spec.
+    ///
+    /// The most important field of the configuration is the chain specification.
+    /// This is a JSON document containing all the information necessary for the client to
+    /// connect to said chain.
+    ///
+    /// The chain spec must be obtained from a trusted entity.
+    ///
+    /// It can be fetched from a trused node with the following command:
+    /// ```bash
+    /// curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "sync_state_genSyncSpec", "params":[true]}' http://localhost:9944/ | jq .result > res.spec
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// For testing environments, please populate the "bootNodes" if the not already provided.
+    /// See [`Self::bootnodes`] for more details.
+    ///
+    /// ```json
+    ///   "bootNodes": [
+    ///       "/ip4/127.0.0.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp"
+    ///    ],
+    /// ```
```

Comment:
I think we should just point at `Self::bootnodes` here, and then if you want, the docs for that function can say something like "setting bootnodes here is equivalent to adding them to the chain spec like so"